samples/menus/nav-drawer/add-drawer-items/src/index.ts

```diff
@@ -5,6 +5,28 @@ import 'igniteui-webcomponents/themes/bootstrap.css';
 defineComponents(IgcNavDrawerComponent, IgcNavDrawerHeaderItemComponent, IgcNavDrawerItemComponent, IgcIconComponent);
 export class NavDrawerAddItems {
     constructor() {
+      const navDrawer = document.querySelector('igc-nav-drawer') as IgcNavDrawerComponent;
+
+      navDrawer.onclick = (e) => {
+        const target = e.target as HTMLElement;
+        const drawerItem = target.closest('igc-nav-drawer-item');
+
+        if (drawerItem) {
+          const iconName = drawerItem.querySelector('igc-icon')?.name;
+          const icons = document.querySelectorAll(`igc-icon`);
+
+          icons.forEach(icon => {
+            const parentItem = icon.parentElement! as IgcNavDrawerItemComponent;
+
+            if(icon.name === iconName) {
+              parentItem.active = true;
+            } else {
+              parentItem.active = false;
+            }
+          })
```

Comment:
We may just match the drawer items by reference, there is no need to compare their icons.